From kay.sievers@vrfy.org Thu Apr  5 16:41:20 2007
From: Kay Sievers <kay.sievers@vrfy.org>
Subject: Driver core: warn when userspace writes to the uevent file in a non-supported way
To: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 06 Apr 2007 01:40:38 +0200
Message-Id: <1175816438.4092.1.camel@lov.localdomain>

In the future we will allow the uevent type to be written to the uevent
file to trigger the different types of uevents.  But for now, as we only
support the ADD event, warn if userspace tries to write anything else to
this file.

Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/base/core.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -296,6 +296,9 @@ out:
 static ssize_t store_uevent(struct device *dev, struct device_attribute *attr,
 			    const char *buf, size_t count)
 {
+	if (memcmp(buf, "add", 3) != 0)
+		dev_err(dev, "uevent: unsupported action-string; this will "
+			"be ignored in a future kernel version");
 	kobject_uevent(&dev->kobj, KOBJ_ADD);
 	return count;
 }
